scripts/install_appenginesdk.py

```diff
@@ -0,0 +1,1400 @@
+
+    def test_main_101_download_(self):
+        self.assertReturncode(main, ['--clean', '--download'])
+
+    def test_main_102_clean_download_unzip(self):
+        self.assertReturncode(main, ['--unzip'])
+        with self.assertRaises(Exception):
+            self.assertReturncode(main, ['--unzip'])
+        self.assertReturncode(main, ['--clean', '--unzip'])
+
+    def test_main_103_print_version(self):
+        self.assertReturncode(main, ['--print-version'])
+        self.assertReturncode(main, ['--print-version', '-q'])
+        self.assertReturncode(main, ['-s', 'print_version'])
+
+    def test_main_104_print_env(self):
+        self.assertReturncode(main, ['--print-env'])
+        self.assertReturncode(main, ['--print-env', '-q'])
+        self.assertReturncode(main, ['-s', 'print_env'])
+
+    def test_main_105_print_env_all(self):
+        self.assertReturncode(main, ['--print-env-all'])
+        self.assertReturncode(main, ['--print-env-all', '-q'])
+        self.assertReturncode(main, ['-s', 'print_env_all'])
+
+    def test_main_106_check(self):
+        self.assertReturncode(main, ['--check'])
+        self.assertReturncode(main, ['-s', 'check'])
+
+    def test_main_107_empty_step_err(self):
+        with self.assertRaises(SystemExit):
+            self.assertReturncode(main, ['-s'])
+
+
+def main(argv=None, stdout=sys.stdout, stderr=sys.stderr, cfg=None):
+    import optparse
+    prs = optparse.OptionParser(
+        usage="%prog [--config] [--install] [--download|--unzip]")
+    prs.add_option('--check',
+                   dest='check',
+                   action='store_true',
+                   help='Check AppEngine SDK installation')
+    prs.add_option('--install',
+                   dest='install',
+                   action='store_true',
+                   help='Install AppEngine SDK [--clean, --download, --unzip]')
+    prs.add_option('--clean',
+                   dest='clean',
+                   action='store_true',
+                   help="Remove existing APPENGINESDK_PREFIX")
+    prs.add_option('--download',
+                   dest='download',
+                   action='store_true',
+                   help='Download AppEngine SDK')
+    prs.add_option('--unzip',
+                   dest='unzip',
+                   action='store_true',
+                   help='Unzip AppEngine SDK in APPENGINESDK_BASEPATH')
+
+    prs.add_option('-s', '--step',
+                   dest='steps',
+                   action='append',
+                   help='Run a named step (see: --list-steps)')
+
+    prs.add_option('--environ',
+                   dest='config_from_environ',
+                   action='store_true',
+                   help='Read environment variables from os.environ into cfg')
+
+    prs.add_option('--version',
+                   dest='APPENGINESDK_VERSION',
+                   action='store',
+                   default=None,
+                   help="APPENGINESDK_VERSION to install")
```

Comment:
- [ ] SEC,ENH: determine what the remote url for VERSION.yaml is from appenginesdk .py